loadgen/loadgen.cc

```diff
@@ -318,9 +318,7 @@ std::vector<QueryMetadata> GenerateQueries(
 
   // When sample_concatenate_permutation is turned on, pad to a multiple of the
   // complete dataset to ensure complete fairness.
-  // FIXME: Only override this for Offline; fix after v2.0
   if (settings.sample_concatenate_permutation &&
-      scenario == TestScenario::Offline &&
```

Comment:
I think we should keep this line as if we remove this, samples_per_query will be (multiple of) sample set size for any other scenario like SingleStream or Server. Let's keep this line.

Reply:
Done.